includes/imemory.hpp

```diff
@@ -14,8 +16,46 @@ class IMemory
     using CartridgeData = std::array<uint8_t, cartridgeSize>;
     using RomData = std::array<uint8_t, romSize>;
 
-    virtual CartridgeData const & getCartridge() = 0;
-    virtual RomData const & getReadOnlyMemory() = 0;
+    enum class REG8BIT
+        {
+            A, F,
+            B, C,
+            D, E,
+            H, L
+        };
+
+    enum class REG16BIT
+        {
+            AF,
+            BC,
+            DE,
+            HL,
+            SP,
+            PC
+        };
+
+    enum class FLAG
+        {
+            Z = 7,
```

Comment:
are you sure it's 7 and not 0x7 ?

Reply:
This is the number of the bit which represent the flag, so no need for hex here